### PR TITLE
feat: slides implementation

### PR DESCRIPTION
**Pull request by kylengn**

### CODE REVIEW

File: package/components/presentation-mode/preview-panel.tsx [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+    const isIOS = checkOs() === 'iOS';
+    const isMobile = useMediaQuery('(max-width: 1280px)');
+
+    const scrollToCurrentSlide = (slideIndex: number) => {
+        if (!previewPanelRef.current || !slideRefs.current[slideIndex]) return;
+
+        const panel = previewPanelRef.current;
+        const slideElement = slideRefs.current[slideIndex];
+
+        if (slideElement) {
```
Comment (nadeem-fileverse) [inline package/components/presentation-mode/preview-panel.tsx:33]: We can simplify this nesting `if(!slideElement) return`

File: package/extensions/mardown-paste-handler/index.ts [hunk 2]
```diff
@@ -379,13 +427,52 @@ function handleMarkdownContent(view: any, content: string) {
...
   const paragraphs = doc.getElementsByTagName('p');
   for (let i = paragraphs.length - 1; i >= 0; i--) {
     const p = paragraphs[i];
     if (p.childNodes.length === 1 && p.firstChild?.nodeName === 'IMG') {
       p.parentNode?.replaceChild(p.firstChild, p);
     }
+    // Replace === text content with a div element that will be converted to a page break node
+    if (
+      p.childNodes.length === 1 &&
+      p.firstChild?.textContent?.trim() === '==='
```
Comment (nadeem-fileverse) [inline package/extensions/mardown-paste-handler/index.ts:469]: This might add page break if textContent in list, tables, headings etc is `===`

Comment (nadeem-fileverse) [reply package/extensions/mardown-paste-handler/index.ts:469]: Not sure if that is fine

**Review by nadeem-fileverse — COMMENTED** [review]
PTAL

File: package/components/presentation-mode/presentation-mode.tsx [hunk 3]
```diff
@@ -0,0 +1,433 @@
...
+            // You'll need to pass an onError prop to show the toast
+            onError?.("Cannot enter presentation mode with empty content");
+            return;
+        }
+
+        setIsLoading(true);
+        const markdown = convertToMarkdown(editor);
+
+        // Add artificial delay of 3 seconds
+        const timeoutId = setTimeout(() => {
```
[the remark this answers is not in the document]
Comment (kylengn) [reply package/components/presentation-mode/presentation-mode.tsx:96]: This is for loading screen of building slides, timeout to have a delay to show the animated loader

**Pull request merged**